soc/arm64/nxp_imx/common/include/soc.h

```diff
@@ -0,0 +1,13 @@
+/*
+ * Copyright 2023 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_SOC_NXP_IMX_COMMON_SOC_H_
+#define ZEPHYR_SOC_NXP_IMX_COMMON_SOC_H_
+
+/* TODO: populate me when need be */
```

Comment:
please, avoid usage of soc.h. It was removed for good reasons from ARM64 a while ago.https://github.com/zephyrproject-rtos/zephyr/pull/46147

Reply:
ACK, will try to see if the issue this PR fixes can be solved from SOF side or try to at least find a better alternative.